csds-client/client/client.go

```diff
@@ -0,0 +1,228 @@
+package client
+
+import (
+	csdspb "github.com/envoyproxy/go-control-plane/envoy/service/status/v2"
+	envoy_type_matcher "github.com/envoyproxy/go-control-plane/envoy/type/matcher"
+
+	"context"
+	"crypto/x509"
+	"errors"
+	"flag"
+	"fmt"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/oauth"
+	"google.golang.org/grpc/metadata"
+	"time"
+)
+
+type Flag struct {
+	uri             string
+	platform        string
+	authnMode       string
+	apiVersion      string
+	requestFile     string
+	requestYaml     string
+	jwt             string
+	configFile      string
+	monitorInterval time.Duration
+}
+
+type Client struct {
+	cc         *grpc.ClientConn
+	csdsClient csdspb.ClientStatusDiscoveryServiceClient
+
+	nm   []*envoy_type_matcher.NodeMatcher
+	md   metadata.MD
+	info Flag
+}
+
+// ParseFLags parses flags to info
+func ParseFlags() Flag {
+	uriPtr := flag.String("service_uri", "trafficdirector.googleapis.com:443", "the uri of the service to connect to")
```

Comment:
May I extract these default values out into constants in the refactor PR? Since I've planned to move the `ParseFlags()` to main to make the client more reusable in that PR and it would be better to extract these to constants there.